### PR TITLE
Give metrics and logging a verdict dimension

### PR DESCRIPTION
**Pull request by paulgmiller**
Playing around with testing and wanted have a little more info on packets per verdict

### CODE REVIEW

**Comment by aojea** [comment]
/ok-to-test
Thanks

File: pkg/networkpolicy/metrics.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ var (
 	packetCounterVec = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Name: "packet_count",
 		Help: "Number of packets",
-	}, []string{"protocol", "family"})
+	}, []string{"protocol", "family", "verdict"})
```
Comment (paulgmiller) [inline pkg/networkpolicy/metrics.go:36]: If we tossed the tossed the whole packet 5 part tuple in here,  we  could use a cli tool to generate network policies for the last amount of time all your daemonsets were up. But that seem like maybe an optional metric to leave on so leaving for another PR.

File: pkg/networkpolicy/controller.go [hunk 2]
```diff
@@ -458,8 +458,9 @@ func (c *Controller) Run(ctx context.Context) error {
 			processingTime := float64(time.Since(startTime).Microseconds())
 			packetProcessingHist.WithLabelValues(string(packet.proto), string(packet.family)).Observe(processingTime)
 			packetProcessingSum.Observe(processingTime)
-			packetCounterVec.WithLabelValues(string(packet.proto), string(packet.family)).Inc()
-			klog.V(2).Infof("Finished syncing packet %d took: %v accepted: %v", *a.PacketID, time.Since(startTime), verdict == nfqueue.NfAccept)
+			verdictStr := verdictString(verdict)
+			packetCounterVec.WithLabelValues(string(packet.proto), string(packet.family), verdictStr).Inc()
+			klog.V(2).Infof("Finished syncing packet %d took: %v verdict: %v", *a.PacketID, time.Since(startTime), verdictStr)
```
Comment (paulgmiller) [inline pkg/networkpolicy/controller.go:463]: This conflicts with https://github.com/kubernetes-sigs/kube-network-policies/pull/50 so will wait on @uablrek if that's inbound

Comment (uablrek) [reply pkg/networkpolicy/controller.go:463]: I re-based #50

Comment (aojea) [reply pkg/networkpolicy/controller.go:463]: should we close this now that #50 merged?

**Comment by paulgmiller** [comment]
This seems to be working as intended when I toss it on a cluster

# TYPE packet_count counter
packet_count{family="IPv4",protocol="TCP",verdict="accept"} 627
packet_count{family="IPv4",protocol="TCP",verdict="drop"} 13

File: pkg/networkpolicy/controller.go [hunk 3]
```diff
@@ -493,6 +494,19 @@ func (c *Controller) Run(ctx context.Context) error {
...
+// verifctString coverts nfqueue int vericts to strings for metrics/logging
+// it does not cover all of them because we should only use a subset.
+func verdictString(verdict int) string {
+	switch verdict {
+	case nfqueue.NfDrop:
+		return "drop"
+	case nfqueue.NfAccept:
+		return "accept"
+	default:
+		return "unknonw"
```
Comment (aojea) [inline pkg/networkpolicy/controller.go:506]:
```suggestion
		return "unknown"
```

**Comment by aojea** [comment]
one typo to fix and ready to merge

File: pkg/networkpolicy/metrics.go (hunk 1 above)
Comment (paulgmiller) [reply pkg/networkpolicy/metrics.go:36]: Yep exactly and it doesn't even catch the labels so would almost certainly have to come through logs to build something like this. 

Verdict is still helpful though as people can alert on continuos blocking as its either a misconfiguration or an attack.

Comment (aojea) [reply pkg/networkpolicy/metrics.go:36]: agree

**Comment by aojea** [comment]
/lgtm
/approve

Thanks
